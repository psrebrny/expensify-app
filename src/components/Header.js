import React from 'react';
import { NavLink } from 'react-router-dom';

export const Header = () => {
  return (
      <div>
        <h1>Expensify</h1>
        <NavLink exact to="/" activeClassName="is-active">Home</NavLink><br/>
        <NavLink to="/create" activeClassName="is-active">Create</NavLink><br/>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
      </div>
  );
};