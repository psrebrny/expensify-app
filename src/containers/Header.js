import React from 'react';
import { Link } from 'react-router-dom';
import { logout } from '../actions/auth';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

export const HeaderClass = (props) => {
  return (
      <header className="header">
        <div className="content-container">
          <div className="header__content">
            <Link className="header__title" to="/dashboard">
              <h1>Expensify</h1>
            </Link>
            <button className="button button--link" onClick={() => {
              props.logout();
            }}>Logout
            </button>
          </div>
        </div>
      </header>
  );
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    logout
  }, dispatch);
};

export const Header = connect(null, mapDispatchToProps)(HeaderClass);