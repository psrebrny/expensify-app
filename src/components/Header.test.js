import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('components/header.js', () => {
  test('render header', () => {
    // const renderer = new ReactShallowRender();
    // renderer.render(<Header/>);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
    
    const wrapper = shallow(<Header/>);
    // expect(wrapper.find('h1').text()).toBe('Expensify');
    expect(wrapper).toMatchSnapshot();
  });
});