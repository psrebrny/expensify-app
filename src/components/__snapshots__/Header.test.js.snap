// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/header.js render header 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Help
  </NavLink>
</div>
`;
